packages/lodestar-validator/src/services/attestation.ts

```diff
@@ -66,16 +68,24 @@ export class AttestationService {
   }
 
   public start = async (): Promise<void> => {
+    this.controller = new AbortController();
     const slot = this.provider.getCurrentSlot();
-    //get current epoch duties
+    // get current epoch duties
     await this.onNewEpoch(computeEpochAtSlot(this.config, slot) - 1);
 
     if (computeStartSlotAtEpoch(this.config, computeEpochAtSlot(this.config, slot)) !== slot) {
-      //trigger next epoch duties
+      // trigger next epoch duties
       await this.onNewEpoch(computeEpochAtSlot(this.config, slot));
     }
   };
 
+  public stop = async (): Promise<void> => {
+    await this.provider.disconnect();
```

Comment:
I don't think we need that in services (unsubscribe would be okeish but we don't need it). It's fine just to call disconnect in validator class